### PR TITLE
Email verification

### PR DESCRIPTION
**Pull request by nabil-dbz**
- When the user creates an account, a verification email gets sent to their inbox.

- The user gets added to the database, even if the email is not verified, with a property emailVerified that get updated every time they sign in.

-  If the email is not verified, the use can't sign in and their profile page can't be accessed.

### CODE REVIEW

**Review by zhoua0805 — APPROVED** [review]
LGTM

File: planet/src/main/webapp/js/authentication.js [hunk 1]
```diff
@@ -86,12 +95,59 @@ async function generateUsername(displayName) {
...
+        addUserToDatabase(user.uid, displayName, phoneNumber, email, 
+            username, user.emailVerified);
+    })
+    .catch(function(error) {
+        const errorCode = error.code;
+        const errorMessage = error.message;
+        if (errorCode === 'auth/missing-android-pkg-name') {
+            alert('Android package name is missing');
+        } else if (errorCode === 'auth/missing-continue-uri') {
+            alert('Continue URL is missing');
```
Comment (EkkiD) [inline planet/src/main/webapp/js/authentication.js]: what is the user supposed to do about these alerts? If there's no user action that can be taken, we can show the user a more friendly generic error and log the real error

Comment (nabil-dbz) [reply planet/src/main/webapp/js/authentication.js]: No! The user can't do anything about them. It's probably better to just log the error on the console

**Review by EkkiD — APPROVED** [review]
LGTM

**Pull request merged**